tools/cd_scripts/package_gcsfuse.sh

```diff
@@ -0,0 +1,107 @@
+# gcsfuse_RELEASE_VERSION_arm64.deb
+# gcsfuse-RELEASE_VERSION-1.aarch64.rpm
+# gcsfuse-RELEASE_VERSION-1.x86_64.rpm
+
+set -e
+# create-gcsfuse-packages VM is a fixed VM in GCP project gcs-fuse-test
+
+# Function to fetch metadata value of the key.
+function fetch_meta_data_value() {
+  metadata_key=$1
+  # Fetch metadata value of the key
+  gcloud compute instances describe create-gcsfuse-packages --zone us-west1-b --flatten="metadata[$metadata_key]" >>  metadata.txt
+  # cat metadata.txt.txt
+  # ---
+  #   value
+  x=$(sed '2!d' metadata.txt)
+  #   value(contains preceding spaces)
+  # Remove spaces
+  # value
+  var_to_set_value=$(echo "$x" | sed 's/[[:space:]]//g')
+  # echo $var_to_set_value
+  # value
+  rm metadata.txt
+}
+
+function exit_in_docket_build_failure() {
+    if [[ $? -ne 0 ]]; then
+      echo "docker build fails."
+      exit 1
+    fi
+}
+# Fetch metadata value of the key "RELEASE_VERSION"
+fetch_meta_data_value "RELEASE_VERSION"
+RELEASE_VERSION=$var_to_set_value
+echo RELEASE_VERSION="$RELEASE_VERSION"
+
+# Fetch metadata value of the key "UPLOAD_BUCKET"
+fetch_meta_data_value "UPLOAD_BUCKET"
+UPLOAD_BUCKET=$var_to_set_value
+echo UPLOAD_BUCKET="$UPLOAD_BUCKET"
+
+sudo apt-get update
+echo Install docker
+sudo apt install apt-transport-https ca-certificates curl software-properties-common -y
+curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu focal stable"
+apt-cache policy docker-ce
+sudo apt install docker-ce -y
+echo Install git
+sudo apt-get install git -y
+# It is require for multi-arch support
+sudo apt-get install qemu-user-static binfmt-support
+
+git clone https://github.com/GoogleCloudPlatform/gcsfuse.git
+cd gcsfuse/tools/package_gcsfuse_docker/
+
+# Setting set +e to capture error output in log file and send it on the bucket.
+set +e
+echo "Building docker for amd64 ..."
+sudo docker buildx build --load . -t gcsfuse-release-amd64:"$RELEASE_VERSION" --build-arg GCSFUSE_VERSION="$RELEASE_VERSION" --build-arg ARCHITECTURE=amd64 --platform=linux/amd64 &> docker_amd64.log &
+pid1=$!
+echo "Building docker for arm64 ..."
+# It is necessary for cross-platform image building because it creates a builder instance that is capable of
+# building images for multiple architectures.
+sudo docker buildx create --name mybuilder --bootstrap --use
+sudo docker buildx build --load . -t gcsfuse-release-arm64:"$RELEASE_VERSION" --build-arg GCSFUSE_VERSION="$RELEASE_VERSION" --build-arg ARCHITECTURE=arm64 --platform=linux/arm64 &> docker_arm64.log &
+pid2=$!
+echo "Waiting for both builds to complete ..."
+wait $pid1
+gsutil cp docker_amd64.log gs://"$UPLOAD_BUCKET"/v"$RELEASE_VERSION"/
+# Exit if any of the build fails.
+exit_in_docket_build_failure
+wait $pid2
+gsutil cp docker_arm64.log gs://"$UPLOAD_BUCKET"/v"$RELEASE_VERSION"/
+# Exit if any of the build fails.
+exit_in_docket_build_failure
```

Comment:
I believe the $? inside this function would take exit code of "gsutil cp docker_arm64.log gs://"$UPLOAD_BUCKET"/v"$RELEASE_VERSION"/" but we want to take exit code of wait $pid2.

Reply:
Sorry I missed